tiledb/sm/serialization/consolidation.cc

```diff
@@ -113,7 +118,8 @@ array_consolidation_request_from_capnp(
     auto fragment_reader = array_consolidation_request_reader.getFragments();
     fragment_uris.reserve(fragment_reader.size());
     for (const auto& fragment_uri : fragment_reader) {
-      fragment_uris.emplace_back(fragment_uri);
+      fragment_uris.emplace_back(deserialize_array_uri_to_absolute(
```

Comment:
REST CI is failing because the path normalization here is not happening on REST since go-capnp handles deserializing the request from generated source code. So we are serializing absolute URIs as relative URIs from core client-side, but REST does not call `deserialize_consolidation_request` and the switch back to absolute never happens.

We could add a `tiledb_handle_array_consolidation_request` much like `tiledb_handle_load_array_schema_request` that would just call `deserialize_consolidation_request`, but we'll need a TileDB-Go binding and subsequent REST PR to update the route to use it.

Another option without the handler and Go binding is to handle a second type of relative URIs in [FragmentConsolidator::consolidate_fragments](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/consolidator/fragment_consolidator.cc#L366) that is relative to the `array_uri/` directory. The relative URIs handled in `consolidate_fragments` linked above is currently relative to the `array_uri/__fragments` directory.

Reply:
How about reconstructing the absolute URIs in [HandleConsolidateArrayRequest](https://github.com/TileDB-Inc/TileDB-Cloud-REST/blob/797505f313d1df7cbef5a7e9c103bc42062d0fb9/internal/arrays/api/v1/array.go#L524) by prepeding the array URI to `arrayConsolidationRequest.Fragments` before calling [ConsolidateFragments](https://github.com/TileDB-Inc/TileDB-Cloud-REST/blob/797505f313d1df7cbef5a7e9c103bc42062d0fb9/internal/arrays/api/v1/array.go#L615C47-L615C82) ? Would that be possible? Looks like the least "invasive" solution.

Reply:
Opened this to track, we discussed and this is a feasible REST-Server side solution: https://app.shortcut.com/tiledb-inc/story/52679/fix-fragment-list-consolidation-handler-to-convert-relative-uris-to-absolute

Let's get this reviewed in the meantime when you have some spare time.

Reply:
This is done, I merged this branch to fix conflicts. I'll watch for CI failures but LGTM :+1: